docs/release_process.md

```diff
@@ -144,28 +141,54 @@ After you push the tag to GitHub, you should also
    - Input the tag created earlier (today's date, e.g., `2024.09.01`)
    - Click on `Run workflow` when ready
 7. A webhook will send status messages into the `#fx-monitor-engineering` channel.
-   - Watch for messages: `pushing to production started` and `successfully deployed to production`
 8. After successful deploy, conduct some basic sanity check:
    - Check sentry prod project for a spike in any new issues
    - Check [grafana dashboard][grafana-dashboard] for any unexpected spike in ops
    - Spot-check the site for basic functionality
 
+Note the following caveats:
+
+- All production releases must follow the date tag naming scheme or else they will not be deployed. The regex for image deployments is in the [monitor tenant definition file](https://github.com/mozilla/global-platform-admin/blob/main/tenants/monitor.yaml).
+  - Example: `2024.09.01`; also can include additional numeric suffix preceded by a period, e.g. `2024.09.01.1` (for manually created deployments, typically hotfixes)
+- All dev releases must use the short commit SHA as the tag or else they will not be deployed
+- Only commits which have already been deployed to staging can be released to development or production environments. In practice this means that you cannot and should not attempt to deploy from a tag created on a release branch
+<!-- 
+Note: The reason that this is the case currently is due to a race condition with the on-push triggered jobs release_retag_v2 and docker_build_deploy_v2. The release_retag_v2 action attempts to pull an image uploaded by docker_build_deploy_v2 immediately, even though the job is not complete. However, the new tag _will_ create a staging release (tagged by the short commit SHA) that will override the current staging environment. This could result in confusion about the state of staging environment and commit verification by QA.
+
+A prod release could be forced by re-running the failed release_retag_v2 job after docker_build_deploy_v2 completes, but this is not recommended. If the changes in this release branch are not merged into main and included in subsequent releases, then they could be unintentionally reverted.
```

Comment:
```suggestion
A prod release could be forced by re-running the failed release_retag_v2 job after docker_build_deploy_v2 completes, but this should be the exception, and you should make sure to merge the release branch back into `main` afterwards to prevent it from being unintentionally reverted.

Also note that any migrations already in `main` should also be included in your release branch.
```

(We've had a number of cases where this would definitely have been safer than just deploying the latest `main`, or than reverting and re-applying other changes.)